### PR TITLE
Create validation package

### PR DESCRIPTION
**Pull request by faihegberg**
Implements a validation package to replace some of our `assert` and `fail` usages.

Part of https://github.com/act-now-coalition/act-now-packages/issues/561

To test:
1. Checkout this PR and run storybook.
2. In `MetricLegendCategorical`, change the metric of any of the stories to `MetricId.PI`. The story should throw an `ActNowJsError` with the appropriate error message.

### CODE REVIEW

File: src/validate/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  });
+});
+
+describe("throwActNowJsError", () => {
+  const errorMsg = "Error message";
+  test("throws an error with the given message", () => {
+    expect(() => throwError()).toThrow();
+    expect(() => throwError(errorMsg)).toThrowError(errorMsg);
+  });
+});
```
Comment (faihegberg) [inline src/validate/index.test.ts]: These tests are the same as the ones for `assert`

File: src/validate/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# @actnowcoalition/validate
+
+Validation functions
+
+## Installing
+
+```sh
+yarn add @actnowcoalition/validate
```
Comment (mikelehen) [inline src/validate/README.md]: Ooops, now that we have merged our packages into just `@actnowcoalition/actnow.js` this README.md doesn't make much sense (there's no separate step to install individual packages).  So I'd probably just delete it.

File: src/validate/index.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+): asserts condition {
+  if (!condition) {
+    throw new ActNowJsError(errorMessage);
+  }
+}
+
+/**
+ * Throws an Error with the (optionally) provided error message.
+ */
+export function throwError(errorMessage?: string): never {
```
Comment (mikelehen) [inline src/validate/index.ts]: I'd put `validation` in the name.  `throwValidationError()` or `failValidation()` or something.

**Review by mikelehen — APPROVED** [review]
Looks great! 🙌

File: src/validate/README.md (hunk 2 above)
Comment (faihegberg) [reply src/validate/README.md]: Ah right, thanks for the catch! 😅

**Pull request merged**